### PR TITLE
US-61

### PR DESCRIPTION
**Pull request by cghende1**
Testing Procedure:
1. Create a new project and verify that the new project dialog allows you to enter working hours 0-24 for each day of the week.
2. Go to your memoranda save data directory (on Windows it is C:\Users*YourUsername*.memoranda), open the .projects file with 
   your text editor of choice, and verify that your newly-created project has a workinghrs property with the corresponding values
   you entered for the weekly working hours.
3. Set the new project you created (the one for which you entered working hours) as the current project, then choose File > Pack project… 
   and save the project as a .zip file.  Next, delete the project from Memoranda, then re-import it using File > Unpack project…  Finally, 
   open the .projects file in the .memoranda directory and verify that the project has been properly re-imported with the same values for 
   workinghrs that you had entered previously.

WARNING: This new code assumes that all new projects will have been created with WorkingHours, attempting to pack or unpack old projects 
         created without WorkingHours will cause problems!  If need be you can delete your .memoranda directory and then Memoranda will 
         recreate it the next time it is started, thus ensuring that all projects will be created with WorkingHours.

### CODE REVIEW

**Comment by cghende1** [comment]
There have been a lot of commits to this PR. Is it ready for review or are you still working on it?

**Comment by jgerszty** [comment]
I get [this](https://gist.github.com/jgerszty/730a8a5b33eed881e85c) error when I try to pack the project without restarting. Works fine after shutting down the program.

**Comment by christopher-henderson** [comment]
Jacob, has your error been resolved? Can we merge this?

**Comment by jgerszty** [comment]
I'm still getting the error if I don't restart Memoranda before trying to pack.

**Comment by cghende1** [comment]
Confirmed on Mac.

File: src/net/sf/memoranda/ProjectImpl.java [hunk 1]
```diff
@@ -97,6 +98,18 @@ else if (today.after(prEnd)) {
         else
             return Project.SCHEDULED;
     }
+    
+    
+    public WorkingHours getWorkingHours(){
+    	Attribute d = _root.getAttribute("workingHrs");
+        if (d == null) return null;
```
Comment (cghende1) [inline src/net/sf/memoranda/ProjectImpl.java]: Please move this to two lines with braces.

File: src/net/sf/memoranda/ProjectImpl.java [hunk 2]
```diff
@@ -97,6 +98,18 @@ else if (today.after(prEnd)) {
...
+        return new WorkingHours(d.getValue());    
+    }
+    
+    public void setWorkingHours(WorkingHours hrs){
+    	if (hrs != null)
```
Comment (cghende1) [inline src/net/sf/memoranda/ProjectImpl.java]: Please use braces.

File: src/net/sf/memoranda/util/Util.java [hunk 3]
```diff
@@ -81,6 +82,16 @@ public static String getDateStamp(CalendarDate date) {
         int[] ret = {d.getDay(), d.getMonth(), d.getYear()};
         return ret;*/
     }
+    
+    public static int[] parseWorkingHrsStamp(String s){
+    	s = s.trim();
+    	Scanner scan = new Scanner(s);
+    	int[] hrs = new int[7];
+    	for(int i = 0; i < 7; i++)
```
Comment (cghende1) [inline src/net/sf/memoranda/util/Util.java]: Please add in braces.

**Comment by jgerszty** [comment]
Looks good, pliggity plus 1.

File: src/net/sf/memoranda/workinghrs/WorkingHours.java [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+	public int getSaturdayHours(){
+		return sat;
+	}
+	public int getSundayHours(){
+		return sun;
+	}
+	
+	//Setters
+	public void setMondayHours(int hours) throws IllegalArgumentException {
+		if(hours >= 0  && hours <= 24)
```
Comment (cghende1) [inline src/net/sf/memoranda/workinghrs/WorkingHours.java]: Braces for alllllll of these.

**Comment by cmbartee** [comment]
Fixed the issues with the lack of enclosing braces that cghende1 pointed out.

**Comment by cghende1** [comment]
Looks good now after an olympic rebase.

**Comment by crdiaz3** [comment]
+1

**Comment by jgerszty** [comment]
:+1: :100:

**Comment by ajcallos** [comment]
Acceptance test passes

**Pull request merged**